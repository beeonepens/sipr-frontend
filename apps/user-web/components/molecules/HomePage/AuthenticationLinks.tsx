import { Button } from 'ui';
import LinkTo from '@components/atoms/LinkTo';
import * as React from 'react';

export default function AuthenticationLinks() {
  return (
    <div className="mx-auto grid w-full grid-cols-1 items-center justify-center gap-8 px-4 md:w-96 md:grid-cols-2">
      <LinkTo to="/register">
        <Button fullWidth variant="outline">
          Register
        </Button>
      </LinkTo>
      <LinkTo to="/login">
        <Button fullWidth>Login</Button>
      </LinkTo>
    </div>
  );
}
