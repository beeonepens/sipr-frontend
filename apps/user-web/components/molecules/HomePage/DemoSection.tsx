import { useState } from 'react';
import DemoSectionModal from './DemoSection/DemoSectionModal';

export default function DemoSection() {
  const [isModalOpen, setIsModalOpen] = useState(false);

  const toggleModal = () => setIsModalOpen((cval) => !cval);

  return (
    <>
      <div className="mt-2">
        <p className="text-base text-gray-700 dark:text-gray-300">
          Or{' '}
          <button
            type="button"
            onClick={toggleModal}
            className="text-primary-600 dark:text-primary-400 cursor-pointer hover:underline"
          >
            click here
          </button>{' '}
          to try the demo without signing up!
        </p>
      </div>

      <DemoSectionModal isModalOpen={isModalOpen} toggleModal={toggleModal} />
    </>
  );
}
