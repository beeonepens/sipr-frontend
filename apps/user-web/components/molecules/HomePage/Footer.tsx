import Logo from '@components/atoms/Logo';
import { useLogoColor } from '@utils/hooks/useLogoColor';

export default function Footer() {
  const logoColor = useLogoColor();
  return (
    <>
      <hr className="opacity-100 dark:opacity-30" />
      <div className="grid grid-cols-none bg-white dark:bg-gray-800 md:grid-cols-2 md:grid-rows-none">
        <div className="md:py-auto hidden px-4 md:mx-24 md:my-auto md:flex">
          <Logo color={logoColor} />
        </div>
        <div className="grid grid-cols-3 py-5 text-center md:text-left">
          <div className="P-5">
            <p className="text-primary-700 dark:text-primary-300 font-bold">
              Features
            </p>
            <ul className="text-primary-700 dark:text-primary-300 mt-4 text-sm">
              <li>Creating Meet</li>
              <li>Join Team</li>
              <li>Meet Invitation</li>
              <li>Notification Agenda</li>
              <li>Documentation</li>
            </ul>
          </div>
          <div className="P-5">
            <p className="text-primary-700 dark:text-primary-300 font-bold">
              Company
            </p>
            <p className="text-primary-700 dark:text-primary-300 mt-4 text-sm">
              PENS
            </p>
          </div>
          <div>
            <p className="text-primary-700 dark:text-primary-300 font-bold">
              About
            </p>
            <ul className="text-primary-700 dark:text-primary-300 mt-4 text-sm">
              <li>About mslis</li>
              <li>Github</li>
              <li>Contact</li>
              <li>Our Team</li>
            </ul>
          </div>
        </div>
      </div>
      <div className="bg-white dark:bg-gray-800">
        <hr className="opacity-100 dark:opacity-30" />
        <p className="text-primary-700 dark:text-primary-300 my-4 text-center font-bold">
          2022 mslis
        </p>
      </div>
    </>
  );
}
