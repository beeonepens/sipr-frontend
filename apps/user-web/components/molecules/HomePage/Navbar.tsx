import Logo from '@components/atoms/Logo';
import { useLogoColor } from '@utils/hooks/useLogoColor';
import AuthenticationLinks from '@components/molecules/HomePage/AuthenticationLinks';

export default function Navbar() {
  const logoColor = useLogoColor();
  return (
    <nav className="bg-white">
      <div className="flex justify-between py-4">
        <div className="flex-shrink-0 px-4">
          <Logo color={logoColor} />
        </div>
        <div className="hidden md:block">
          <div className="flex space-x-4">
            <a
              href="/"
              className=" rounded-md px-3 py-2 text-sm font-bold text-blue-700 hover:text-blue-400"
            >
              Individuals
            </a>

            <a
              href="/"
              className="rounded-md px-3 py-2 text-sm font-bold text-blue-700 hover:text-blue-400"
            >
              Team
            </a>

            <a
              href="/"
              className="rounded-md px-3 py-2 text-sm font-bold text-blue-700 hover:text-blue-400"
            >
              Product
            </a>

            <a
              href="/"
              className="rounded-md px-3 py-2 text-sm font-bold text-blue-700 hover:text-blue-400"
            >
              About
            </a>
          </div>
        </div>
        <div className="items-center">
          <AuthenticationLinks />
        </div>
      </div>
    </nav>
  );
}
