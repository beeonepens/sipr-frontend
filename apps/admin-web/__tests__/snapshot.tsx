import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import HomePage from '../pages';

it('renders homepage unchanged', () => {
  const { container } = render(<HomePage />);
  expect(container).toMatchSnapshot();
});
