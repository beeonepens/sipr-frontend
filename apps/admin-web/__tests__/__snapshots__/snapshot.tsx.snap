// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <article
    class="bg-primary-50 dark:bg-primary-950 py-16"
  >
    <h3
      class="text-center text-2xl font-bold text-gray-800 dark:text-gray-100"
    >
      Welcome to
    </h3>
    <h1
      class="text-primary-900 dark:text-primary-100 mt-1 mb-8 text-center text-4xl font-bold"
    >
      SIPR Admin Page
    </h1>
    <div
      class="flex flex-row justify-center"
    >
      <a
        class="hover:text-primary-500 text-primary-700 underline"
        href="/login"
      >
        <button
          class="border-2 py-2.5 px-6 transition duration-100 ease-in bg-primary-950 border-primary-950 dark:bg-primary-300 dark:hover:bg-primary-400 dark:text-primary-950 hover:bg-primary-850 hover:border-primary-850 dark:border-primary-300 dark:hover:border-primary-400 text-white rounded-md text-base font-medium"
          type="button"
        >
          Login as Admin
        </button>
      </a>
    </div>
  </article>
</div>
`;
